modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/AwaitTasksCompletionExecutor.java

```diff
@@ -50,9 +50,9 @@ class AwaitTasksCompletionExecutor implements Executor {
     public void execute(Runnable command) {
         CompletableFuture<?> future = new CompletableFuture<>();
 
-        future.whenComplete((o, throwable) -> updateHeartbeat.run());
+        CompletableFuture<?> heartbeatFuture = future.whenComplete((o, throwable) -> updateHeartbeat.run());
 
-        pendingTaskFutures.add(future);
+        pendingTaskFutures.add(heartbeatFuture);
```

Comment:
```suggestion
        pendingTaskFutures.add(future.whenComplete((o, throwable) -> updateHeartbeat.run()));
```

Reply:
Why do you think I should inline this statement? We would have 3 methods call in a single line, that's a lot and kinda unnecessary

Reply:
It's just an offer, you can reject it

Reply:
Ok, I just wanted to know your motivation

Reply:
Just fewer lines of code